server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -717,6 +717,19 @@ public static IndexMergePolicy fromString(String text) {
         Property.IndexScope
     );
 
+    /**
+     * Index setting describing the maximum number of uncommitted translog files at a time.
+     * If breached the shard will be Refreshed, this is to control large number of tranlog files
+     * downloads during recovery.
+     */
+    public static final Setting<Integer> INDEX_MAX_UNCOMMITTED_TRANSLOG_FILES = Setting.intSetting(
+        "index.max_uncommitted_translog_files",
+        300,
```

Comment:
Is there any test run to determine the 300 number?

Reply:
@astute-decipher  ^

Reply:
Yes, need to discuss on the minimum and default value here.